src/Microsoft.VisualStudio.Services.Agent/HostContext.cs

```diff
@@ -260,6 +268,26 @@ public virtual string GetDirectory(WellKnownDirectory directory)
             return path;
         }
 
+        public string GetDiagDirectory(HostType hostType = HostType.Undefined)
+        {
+            return hostType switch
+            {
+                HostType.Undefined => GetDiagDirectory(_hostType),
+                HostType.Agent => GetDiagOrDefault(AgentKnobs.AgentDiagLogPath.GetValue(this).AsString()),
+                HostType.Worker => GetDiagOrDefault(AgentKnobs.WorkerDiagLogPath.GetValue(this).AsString()),
+                _ => throw new NotSupportedException($"Unexpected host type: '{hostType}'"),
+            };
+        }
+
+        private string GetDiagOrDefault(string diagFolder)
+        {
+            if (string.IsNullOrEmpty(diagFolder))
+            {
+                diagFolder = new DirectoryInfo(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location)).Parent.FullName;
+            } 
+            return Path.Combine(diagFolder, Constants.Path.DiagDirectory);
```

Comment:
Nit: could it make sense to set it as diagFolder here and hence use diagFolder as '_diag' for related type of logs  - to avoid additional nested level?

Reply:
@anatolybolshakov I don't think I understand how you mean?

Reply:
@teapothat currently agent saves logs in <AGENT_DIAGLOGPATH>/_diag or <WORKER_DIAGLOGPATH>/_diag if any of new knobs are specified. As I undestand '_diag' folder was intended to be just one of the folders in the root agent directory - could it make sense probably to save logs in <AGENT_DIAGLOGPATH> or <WORKER_DIAGLOGPATH> directly instead?
(Hence just omit Path.Combine in line 288 and return `diagFolder` value)

Reply:
@anatolybolshakov So I've updated to not return _diag at the end of the value if it is set from the environment variable. However, the Path.Combine is still needed for the default case.